openpype/pipeline/workfile/workfile_template_builder.py

```diff
@@ -521,10 +521,6 @@ def populate_scene_placeholders(self, level_limit=None):
         if not level_limit:
             level_limit = 1000
 
-        placeholder_by_scene_id = {
```

Comment:
We should fix the issue without getting rid of the recusive functionality.

Reply:
The issue for now is that the recursive functionality is importing assets multiple times. I don't really understand the point of this functionality and its functioning. Could you explain it to me briefly ?

Reply:
Few clients are using it in a way that they publish template workfile (or nodes). Then they define main template which has placehodlers that will load them. So first iteration would load other template placeholders and next iterations would load the placeholders from the previously loaded placeholders. We can't remove it...

Reply:
Just looked at the code and if the `scene_identifier` of placeholder does not change after process, it should not process it again.

Reply:
Ok I see. I will dig into it but I think the `scene_identifier` changes even though it shouldn't. Thank you !